pg_bm25/src/index_access/build.rs

```diff
@@ -127,35 +98,22 @@ unsafe extern "C" fn build_callback(
 unsafe extern "C" fn build_callback_internal(
     ctid: pg_sys::ItemPointerData,
     values: *mut pg_sys::Datum,
-    state: *mut std::os::raw::c_void,
+    _state: *mut std::os::raw::c_void,
     index: pg_sys::Relation,
 ) {
     check_for_interrupts!();
 
-    let index_relation_ref = unsafe { PgRelation::from_pg(index) };
+    let index_relation_ref: PgRelation = PgRelation::from_pg(index);
     let tupdesc = lookup_index_tupdesc(&index_relation_ref);
-    let attributes = categorize_tupdesc(&tupdesc);
-    let natts = tupdesc.natts as usize;
-    let dropped = (0..tupdesc.natts as usize)
-        .map(|i| tupdesc.get(i).unwrap().is_dropped())
-        .collect::<Vec<bool>>();
-
-    let state = (state as *mut BuildState).as_mut().unwrap();
-    let mut old_context = state.memcxt.set_as_current();
```

Comment:
I noticed you got rid of the memory context switching. Just for my understanding, could you talk about why it's safe to do this?

Reply:
We're not using the state anymore in the build callback. Before, we were storing the parade_index on it, but that's not necessary anymore as we just retrieve it from our global cache.